### ISSUE TITLE
Developer documentation - branching strategy

### ISSUE THREAD

**Issue by ccraddock** [onnela-lab/beiwe-backend#75]
Is there a document that describes the purposes of the various branches, or the overall branching strategies? There seem to be a lot of direct commits to master, is it meant to be a current stable version, or somewhat of a development branch? At which point do changes move from master to production? When I setup a new server which of these branches should I use? Have you considered using releases or tags to indicate stable versions?

**Comment by biblicabeebli** [comment]
Branches are currently a mess.  Hooray!

Master is not _intentionally_ used to stage changes or bugfixes, that is what the development branch is _supposed_ to be for, buuut at the moment the python 3 work was immense and iminantly to-be-pushed to production.

Looks like I forgot to bring development up to date with most recent bug fixes that were done on master, I'll go do that.

I just cleared out a bunch of unneeded branches.
There are a couple I've kept around that are from abandoned but potentially useful feature dev, I and some are effectively bugfix branches for transient issues (which I hope to move to cluster management commands).

To answer your question more specifically:
Unless you want to dive into potential issues use the production branch.
Python 3 is about to be pushed to production.
Python 3 is on master because it is a _huge_ code update, all future dev will be python 3, I don't want any contributors doing python 2 codebase edits. (There is already a thread with an issue that might require an update to the python2-legacy branch, uhg.)

Finally, if you have a Python 2 deploy and do not want to upgrade use the python2-legacy branch.
